### ISSUE TITLE
The issue with Thumbnail in TouchableHighlight

### ISSUE THREAD

**Issue by alexn777** [GeekyAnts/NativeBase#574]
Hello,

There is an issue with Thumbnail wrapped by TouchableHighlight. I am using Exponent XDE v14.0.0 and running the project on Genymotion emulator for Google Nexus 5 4.4.4 API 19.

## Versions used
- node.js 7.6.0
- exponent 14.0.0
- react 15.4.0
- react-native 0.41.0
- native-base 2.0.10

## Expected behaviour
I am able to tap Thumbnail wrapped by TouchableHighlight.

## Actual behaviour
When I tap Thumbnail, I get "undefined is not an object (evaluating 'this.wrappedInstance.setNativeProps')".

## The project code
```javascript
import Exponent from 'exponent';
import React from 'react';
import {
  Platform,
  Alert,
  TouchableHighlight
} from 'react-native';
import {
  Container,
  Header,
  Body,
  Title,
  Content,
  Thumbnail
} from 'native-base';

class App extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      isReady: false
    };
  }

  async componentWillMount() {
    if (Platform.OS === 'android') {
      await Exponent.Font.loadAsync({
        'Roboto': require('native-base/Fonts/Roboto.ttf'),
        'Roboto_medium': require('native-base/Fonts/Roboto_medium.ttf')
      });

      this.setState({
        isReady: true
      });
    }
  }

  render() {
    if (!this.state.isReady) {
      return <Exponent.Components.AppLoading />;
    }

    return (
      <Container style={{ flex: 1 }}>
        <Header>
          <Body>
            <Title style={{ alignSelf: 'stretch' }}>Test</Title>
          </Body>
        </Header>
        <Content>
          <TouchableHighlight onPress={ () => Alert.alert('Info', 'Image tapped') }>
            <Thumbnail style={{ width: 50, height: 50 }} source={{ uri: 'http://www.apdconsumo.pt/person.png' }} />
          </TouchableHighlight>
        </Content>
      </Container>
    );
  }
}

Exponent.registerRootComponent(App);
```

## The screenshots
![screenshot1](https://cloud.githubusercontent.com/assets/25953890/23341866/5e2ac15a-fc58-11e6-9d80-b02f9ef89261.png)
![screenshot2](https://cloud.githubusercontent.com/assets/25953890/23341869/658645f0-fc58-11e6-93d9-b020871eddac.png)

## The device
I have tested the project only on Android.

## The additional information
I think, you do not define the `setNativeProps` method in the Thumbnail component. Please, check [Composite components and setNativeProps](https://facebook.github.io/react-native/docs/direct-manipulation.html#composite-components-and-setnativeprops).

**Comment by LeoAref** [comment]
+1

**Comment by sankhadeeproy007** [comment]
I just tested it on iOS and android both and it works. Maybe exponent is causing this?

**Comment by sedders123** [comment]
I also get this error, when the conatiner is wrapped in a style provider

**Comment by sankhadeeproy007** [comment]
![styleprovider](https://cloud.githubusercontent.com/assets/10998047/23410768/3f25dd18-fdf6-11e6-8f5f-aa4254af2f08.gif)
Strange!

**Comment by LeoAref** [comment]
This is the code in my case:
```js
  render() {
    return (
      <View style={{...this.props.style.container, height: this.state.height}}>
        <View onLayout={this.setHeaderHeight}>
          <TouchableHighlight onPress={this.toggle}>
            <Text>click here</Text>
          </TouchableHighlight>
        </View>
        <View onLayout={this.setContentHeight}>
          {this.props.children}
        </View>
      </View>
    );
  }
}

export default connectStyle('zoraTheme.Accordion', styles)(Accordion);
```

And I'm getting the same error

**Comment by alexn777** [comment]
I have just updated native-base to the latest version 2.0.11 and it started working. It did not work with native-base 2.0.10.

**Comment by LeoAref** [comment]
It worked with me as well after install `2.0.11`

**Comment by sedders123** [comment]
+1 to the 2.0.11 fix

**Comment by sankhadeeproy007** [comment]
Great! Closing it then.

**Comment by stonecold123** [comment]
I'm already using 2.0.12, but the issue still comes up. Is it related to react native 0.42? Exactly having the Thumbnail in TouchableHighlight. Anyone?

**Comment by alexn777** [comment]
The issue is back again in native-base 2.0.12.

**Comment by alexn777** [comment]
I get the same error even if I use native-base `Text` in react-native `TouchableHighlight`:

``` javascript
import {
  Alert,
  TouchableHighlight
} from 'react-native';
import {
  Text
} from 'native-base';

<TouchableHighlight onPress={ () => Alert.alert('Info', 'Text touched') }>
  <Text>Click here</Text>
</TouchableHighlight>
```

However, if I use react-native `Text`, it works without issues:

``` javascript
import {
  Alert,
  TouchableHighlight,
  Text
} from 'react-native';

<TouchableHighlight onPress={ () => Alert.alert('Info', 'Text touched') }>
  <Text>Click here</Text>
</TouchableHighlight>
```

**Comment by stonecold123** [comment]
@sankhadeeproy007 , can you reopen the issue? The error comes up again in 2.0.12.

**Comment by abrantes01** [comment]
Same as @stonecold123 , error comes up with `Icon` inside `TouchableHighlight` in 2.0.13.

**Comment by zhoplocal** [comment]
@sankhadeeproy007 the issue was never resolved and it is breaking our projects. please reopen and fix it.

![nbe](https://cloud.githubusercontent.com/assets/26655307/24302354/bbf35678-1088-11e7-9284-e83203603bf1.gif)

**Comment by shivrajkumar** [comment]
We are looking into the issue. As a workaround for now, you can wrap the child component of TouchableHighlight into a RN `<View>`.

**Comment by a11y1337** [comment]
To expand on @shivrajkumar 's workaround, you need to import the View component from React Native: using Native Base's View component still causes the error.

**Comment by sankhadeeproy007** [comment]
Fixed with 2.1.5

**Comment by waqaramjad** [comment]
facing same issue
